spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaAutoConfiguration.java

```diff
@@ -83,110 +87,117 @@
 			"org.hibernate.engine.transaction.jta.platform.internal.WebSphereExtendedJtaPlatform",
 			"org.hibernate.service.jta.platform.internal.WebSphereExtendedJtaPlatform", };
 
-	public HibernateJpaAutoConfiguration(DataSource dataSource,
-			JpaProperties jpaProperties,
-			ObjectProvider<JtaTransactionManager> jtaTransactionManagerProvider) {
-		super(dataSource, jpaProperties, jtaTransactionManagerProvider);
-	}
 
-	@Override
-	protected AbstractJpaVendorAdapter createJpaVendorAdapter() {
-		return new HibernateJpaVendorAdapter();
-	}
+	@Configuration
+	@ConditionalOnSingleCandidate(DataSource.class)
+	static class HibernateJpaConfiguration extends JpaBaseConfiguration {
```

Comment:
That's a bold move but I think I like it. I have a feeling it's going to break user's app, wdyt @wilkinsona ?